tests/std/tests/P0040R3_parallel_memory_algorithms/test.cpp

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <atomic>
+#include <cassert>
+#include <execution>
+#include <iterator>
+#include <memory>
+#include <numeric>
+
+#include <parallel_algorithms_utilities.hpp>
+
+using namespace std;
+using namespace std::execution;
+
+class wrap_uchar {
+private:
+    unsigned char ch_ = 42;
+
+public:
+    bool is_expected() const noexcept {
+        return ch_ == 42;
+    }
+};
+
+static_assert(sizeof(wrap_uchar) == 1);
+
+const auto expectation = [](const wrap_uchar& x) { return x.is_expected(); };
+
+const auto expectation_zero = [](int n) { return n == 0; };
+
+struct test_case_uninitialized_default_construct_parallel {
+    template <typename ExecutionPolicy>

```

Comment:
These template type-parameter introducers should be `class`; we're trying to more closely follow product code conventions in tests.